### PR TITLE
Selector view

### PR DESCRIPTION
**Pull request by rebbrunner**
Front and backend for basic project selection

### CODE REVIEW

File: website/forms.py [hunk 1]
```diff
@@ -45,3 +46,21 @@ class Meta:
             'sharetoken' : 'Allow using token for system jobs'
         }
         widgets = { 'token': forms.TextInput(attrs={'size': 40})}
+
+class ProjectSelectionForm(forms.ModelForm):
+    # pfilter = forms.ModelMultipleChoiceField(Filter.objects.all())
```
Comment (psybers) [inline website/forms.py]: remove if not needed

File: website/templates/website/addFilter.html [hunk 2]
```diff
@@ -0,0 +1,24 @@
+{% extends 'website/layout.html' %}
```
Comment (psybers) [inline website/templates/website/addFilter.html:1]: take a look at the editprofile.html template. forms can be (and should be) just included now.  Mirror that here.

File: website/forms.py [hunk 3]
```diff
@@ -45,3 +46,21 @@ class Meta:
...
+    class Meta:
+        model = ProjectSelector
+        fields = ['input_dataset']
+
+class FilterDetailForm(forms.Form):
+    pfilter = forms.ModelChoiceField(Filter.objects.all())
+    value = forms.CharField(
+        max_length=1000,
+        widget=forms.TextInput(attrs={
+            'class' : 'form-control',
```
Comment (psybers) [inline website/forms.py]: i dont think this has to be set, if you update the template to include the form template (see comment below).

File: website/models.py [hunk 4]
```diff
@@ -93,23 +93,58 @@ class Selection(models.Model):
...
 
     def __str__(self):
-        return self.name
+        return self.value
+
+    def save(self, **kwargs):
+        if self.pfilter.val_type == 'String' and self.value == '':
+            self.value = 'string'
+        elif self.value == '':
+            self.value = ['red', 'green', 'blue']
```
Comment (psybers) [inline website/models.py]: ... huh?

File: website/templates/website/create_normal.html [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+    cloneMore('.form-row:last', 'form');
+    return false;
+});
+$(document).on('click', '.remove-form-row', function(e){
+    e.preventDefault();
+    deleteForm('form', $(this));
+    return false;
+});
+$('select').change(function (){
+    console.log("Hello World!");
```
Comment (psybers) [inline website/templates/website/create_normal.html]: remove debugging messages

File: website/templates/website/editprofile.html [hunk 6]
```diff
@@ -7,10 +7,19 @@
...
+                            {% endfor %}
+                        </ul>
+                    {% endif %}
+                    {{ userForm.as_p }}
+                    {{ profileForm.as_p }}
                     {% include 'website/messages.html' %}
                     {% include 'website/form.html' with form=userForm %}
                     {% include 'website/form.html' with form=profileForm %}
-                    <button type="submit" class="btn btn-primary">Submit</button>
+                    <button type="submit" class="btn btn-default">Submit</button>
```
Comment (psybers) [inline website/templates/website/editprofile.html]: for submit buttons lets use btn-primary (so it turns blue)

File: website/templates/website/editprofile.html [hunk 7]
```diff
@@ -7,10 +7,19 @@
             <div class="panel-body">
                 <form class="form form-default" action="" method="post" enctype="multipart/form-data">
                     {% csrf_token %}
+                    {% if messages %}
```
Comment (psybers) [inline website/templates/website/editprofile.html]: why is this here? you can see im including all this right below it!

File: website/templates/website/layout.html [hunk 8]
```diff
@@ -10,14 +10,13 @@
     {% load staticfiles %}
 
     <!--Bootstrap-->
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
```
Comment (psybers) [inline website/templates/website/layout.html]: viewport is above - remove this

File: website/templates/website/password_reset_email.html [hunk 9]
```diff
@@ -0,0 +1,14 @@
+{% load i18n %}{% autoescape off %}
```
Comment (psybers) [inline website/templates/website/password_reset_email.html]: why is this in this pull request?

File: website/urls.py [hunk 10]
```diff
@@ -15,7 +15,9 @@
     path('logout/', views.logoutView, name='logout'),
     path('register/', views.RegisterView.as_view(), name='register'),
     path('editprofile/', views.profile, name='editProfile'),
-    path('password/reset/', auth_views.PasswordResetView.as_view(email_template_name='website/reset/password_reset_email.html', success_url=reverse_lazy('website:password_reset_done'), template_name='website/reset/passwordReset.html'), name='password_reset'),
+    path('ajax/data_default/', views.data_default, name='data_default'),
+    path('project/selection/', views.project_selection, name='project_selection'),
+    path('password/reset/', auth_views.PasswordResetView.as_view(email_template_name='website/password_reset_email.html', success_url=reverse_lazy('website:password_reset_done'), template_name='website/reset/passwordReset.html'), name='password_reset'),
```
Comment (psybers) [inline website/urls.py]: why was the template renamed/moved?

**Review by psybers — CHANGES_REQUESTED** [review]

File: website/templates/website/password_reset_email.html (hunk 9 above)
Comment (rebbrunner) [reply website/templates/website/password_reset_email.html]: The email template was included because the link's didn't include the website namespace resulting in a url error when Django tried to send emails.

Comment (psybers) [reply website/templates/website/password_reset_email.html]: what do you mean send emails? Why is selector viewing sending emails?

File: website/templates/website/layout.html [hunk 11]
```diff
@@ -11,13 +11,11 @@
 
     <!--Bootstrap-->
     <link rel="stylesheet" href="https://stackpath.bootstrapcdn.com/bootstrap/3.4.0/css/bootstrap.min.css" integrity="sha384-PmY9l28YgO4JwMKbTvgaS7XNZJ30MK9FAZjjzXtlqyZCqBY6X6bXIkM++IkyinN+" crossorigin="anonymous">
-    <script src="https://code.jquery.com/jquery-1.12.4.min.js" integrity="sha384-nvAa0+6Qg9clwYCGGPpDQLVpLNn0fRaROjHqs13t4Ggj3Ez50XnGQqc/r8MhnRDZ" crossorigin="anonymous"></script>
-    <script src="https://stackpath.bootstrapcdn.com/bootstrap/3.4.0/js/bootstrap.min.js" integrity="sha384-vhJnz1OVIdLktyixHY4Uk3OHEwdQqPppqYR8+5mjsauETgLOcEynD9oPHhhz18Nw" crossorigin="anonymous"></script>
+
```
Comment (psybers) [inline website/templates/website/layout.html]: why is selectorview modifying this file?

File: README.md [hunk 12]
```diff
@@ -37,4 +37,4 @@ PAClab development repository using Django framework.
 15. Create static files: `python3 manage.py collectstatic`
```
Comment (psybers) [inline README.md]: why was this file modified?

File: website/admin.py [hunk 13]
```diff
@@ -26,6 +26,15 @@ def get_email(self, obj):
...
 
+class FilterDetailSelectionInline(admin.TabularInline):
+    model = FilterDetail
+    exclude = ['value', 'val_type']
+    extra = 1
+
+class SelectionAdmin(admin.ModelAdmin):
+    inlines = (FilterDetailSelectionInline,)
+
+# Register your models here.
```
Comment (psybers) [inline website/admin.py]: why is this line added?

File: website/forms.py [hunk 14]
```diff
@@ -1,6 +1,7 @@
 from django.contrib.auth.models import User
 from django import forms
-from .models import Profile
+from django.forms import formset_factory
+from .models import Profile, ProjectSelector, Filter, FilterDetail
```
Comment (psybers) [inline website/forms.py]: why were Profile and FilterDetail imported?

File: website/templates/website/create_normal.html [hunk 15]
```diff
@@ -0,0 +1,138 @@
...
+<div class="container">
+    <div class="row"></div>
+    <div class="row">
+        <div class="col-md-12">
+            {% if heading %}
+            <h3>{{heading}}</h3>
+            {% endif %}
+            <form class="form-horizontal" method="POST" action="">
+                {% csrf_token %}
+                {% if messages %}
```
Comment (psybers) [inline website/templates/website/create_normal.html]: include the messages.html template

File: website/templates/website/create_normal.html [hunk 16]
```diff
@@ -0,0 +1,138 @@
+{% extends "website/layout.html" %}
+{% block title %}Project Selection{% endblock %}
+{% block body %}
+<div class="container">
```
Comment (psybers) [inline website/templates/website/create_normal.html:4]: make sure this form mirrors how we handle forms in the form template

File: website/views.py [hunk 17]
```diff
@@ -1,15 +1,16 @@
-from django.http import HttpResponse
+from django.http import HttpResponse, JsonResponse
 from django.shortcuts import render, redirect, get_object_or_404
 from django.views.generic import TemplateView, ListView, DetailView, View
 from django.views.generic.edit import UpdateView, CreateView, DeleteView
 from django.contrib import messages
 from django.contrib.auth import authenticate, login, logout
 from django.contrib.auth.models import User
+from django.contrib.auth.mixins import LoginRequiredMixin
```
Comment (psybers) [inline website/views.py]: this isnt used

File: website/views.py [hunk 18]
```diff
@@ -95,3 +96,47 @@ def profile(request):
...
+                    try:
+                        connection = FilterDetail()
+                        connection.project_selector = ProjectSelector.objects.all().last()
+                        pk = form.cleaned_data.get('pfilter').id
+                        connection.pfilter = Filter.objects.get(pk=pk)
+                        connection.value = form.cleaned_data['value']
+                        connection.save()
+                    except:
+                        pass
+                    # FilterDetail(pfilter=pfilter, value=value).save()
```
Comment (psybers) [inline website/views.py]: remove comments

**Review by psybers — CHANGES_REQUESTED** [review]

File: website/templates/website/layout.html (hunk 11 above)
Comment (rebbrunner) [reply website/templates/website/layout.html]: I think this was the result of me addressing a merge conflict - I remember the bootstrap version was upgraded and it was conflicting with my layout file.

File: README.md (hunk 12 above)
Comment (rebbrunner) [reply README.md]: For some reason my README file on this branch wasn't up to date.

File: website/templates/website/create_normal.html [hunk 19]
```diff
@@ -0,0 +1,132 @@
...
+    <div class="row"></div>
+    <div class="row">
+        <div class="col-md-12">
+            {% if heading %}
```
Comment (psybers) [inline website/templates/website/create_normal.html:8]: why is this conditional? it looks like this template always passes in a heading. Also why are we passing it in instead of just coding it into the template directly?

File: website/templates/website/addFilter.html (hunk 2 above)
Comment (psybers) [reply website/templates/website/addFilter.html:1]: why was this marked resolved but nothing was done about my comment?

File: README.md (hunk 12 above)
Comment (psybers) [reply README.md]: ok so undo the change (whatever it is) here

File: website/templates/website/password_reset_email.html (hunk 9 above)
Comment (psybers) [reply website/templates/website/password_reset_email.html]: i think this file needs removed

**Review by psybers — CHANGES_REQUESTED** [review]

**Review by psybers — APPROVED** [review]
